fcrepo-kernel-modeshape/src/test/java/org/fcrepo/integration/kernel/modeshape/observer/MoveObserverIT.java

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright 2015 DuraSpace, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.fcrepo.integration.kernel.modeshape.observer;
+
+import static java.lang.Thread.sleep;
+import static org.fcrepo.kernel.api.FedoraTypes.FEDORA_CONTAINER;
+import static org.fcrepo.kernel.api.observer.EventType.NODE_ADDED;
+import static org.fcrepo.kernel.api.observer.EventType.NODE_MOVED;
+import static org.fcrepo.kernel.api.observer.EventType.NODE_REMOVED;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+import javax.inject.Inject;
+import javax.jcr.Node;
+import javax.jcr.Repository;
+import javax.jcr.RepositoryException;
+import javax.jcr.Session;
+
+import org.fcrepo.integration.kernel.modeshape.AbstractIT;
+import org.fcrepo.kernel.api.observer.EventType;
+import org.fcrepo.kernel.api.observer.FedoraEvent;
+import org.fcrepo.kernel.api.services.NodeService;
+import org.fcrepo.kernel.modeshape.services.NodeServiceImpl;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.springframework.test.context.ContextConfiguration;
+
+import com.google.common.eventbus.EventBus;
+import com.google.common.eventbus.Subscribe;
+
+import java.util.Set;
+
+/**
+ * <p>MoveObserverIT class: tests to verify that a MOVE operation emits a complete set of repository events.</p>
+ *
+ * @author acoburn
+ */
+@ContextConfiguration({"/spring-test/eventing.xml", "/spring-test/repo.xml"})
+public class MoveObserverIT extends AbstractIT {
+
+    private Integer eventBusMessageCount;
+
+    private Integer removedEvents;
+
+    private Integer movedEvents;
+
+    private Integer createdEvents;
+
+    @Inject
+    private Repository repository;
+
+    @Inject
+    private EventBus eventBus;
+
+    @Test
+    public void TestEventBusPublishing() throws RepositoryException {
+
+        final Session se = repository.login();
+        se.getRootNode().addNode("/object1").addMixin(FEDORA_CONTAINER);
+        se.getRootNode().addNode("/object2").addMixin(FEDORA_CONTAINER);
+        se.save();
+        se.logout();
+
+        try {
+            sleep(500);
```

Comment:
This is not a very strong test. How about using Awaitility to build something a little less brittle?